src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,69 @@
 import React from 'react';
 import './CurrentUser.scss';
+import { getUser } from '../../api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number;
+  selectUser: (id: number) => void;
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  user: User | null;
+  loadingError: boolean;
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    user: null,
+    loadingError: false,
+  };
+
+  componentDidMount() {
+    this.getUser();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.userId !== this.props.userId) {
+      this.getUser();
+    }
+  }
+
+  getUser = () => {
+    getUser(this.props.userId)
+      .then(user => {
+        this.setState({ user });
+      })
+      .catch(()=> {
+        this.setState({ loadingError: true })
+      });
+  };
```

Comment:
```suggestion
  getUser = () => {
  try {
    getUser(this.props.userId)
      .then(user => {
        this.setState({ user });
      });
  } catch (error) {
        this.setState({ loadingError: true })
      });
  };
```
Try this. Let me know whether it fixes the problem

Reply:
I'm not sure whether it is the correct syntax to use 'try-catch' with 'then'... But I'll see what I can do